scripts/proto_splitter/proto2multi.py

```diff
@@ -0,0 +1,172 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Author Simon Steinmann (https://github.com/Simon-Steinmann)
+
+
+# Automatic PROTO to multi-file PROTO trimesh extractor
+#
+# INSTRUCTIONS:
+# Call the script with the --input=<path> argument
+#
+# 1. if <path> ends in <filename.proto>:
+#   - creates a new folder in the same directory named "multifile_<filename>"
+#   - creates a new <filename.proto> inside this folder with all trimeshes replaced
+#      by proto files, placed in a subfolder
+#
+# 2. If <path> does not end in ".proto" -> assumes directory
+#   - creates a new folder inside of <path> called <multi_file_conversion>
+#   - searches for every .proto file in <path> recursively and:
+#       - creates a conversion as described in 1.
+#       - places conversion into the <multi_file_conversion> folder, preserving
+#         the same folder structure as in <path>. This allows <path> to be replaced
+#         with <path>/<multi_file_conversion> when validated, as names structures
+#         are preserved
+#
+
+
+import os
+import optparse
+import errno
+
+
+def mkdirSafe(directory):
+    """Create a dir safely."""
+    try:
+        os.makedirs(directory)
+    except OSError as e:
+        if e.errno != errno.EEXIST:
+            raise
+        else:
+            print('Directory "' + directory + '" already exists!')
+
+
+class proto2multi():
+    def __init__(self):
+        print('Proto 2 multi-file proto converter by Simon Steinmann')
+
+    def header(self, proto):
+        """Specify VRML file header."""
+        proto.write('#VRML_SIM R2020b utf8\n')
+        proto.write('# license: Apache License 2.0\n')
+        proto.write('# license url: http://www.apache.org/licenses/LICENSE-2.0\n')
+        proto.write('# tags: hidden\n')
+        proto.write('# This is a proto file for Webots for the ' + self.robotName + '\n\n')
+
+    def createProto(self, string):
+        name = self.robotName + '_' + str(self.shapeIndex)
+        print('Create meshFile: %sMesh.proto' % name)
+        filepath = '%s/%sMesh.proto' % (self.meshFilesPath, name)
+        meshProtoFile = open(filepath, 'w')
+        self.header(meshProtoFile)
+        meshProtoFile.write('PROTO %sMesh [\n]\n{\n' % name)
+        meshProtoFile.write(string)
+        meshProtoFile.write('}\n')
+        meshProtoFile.close()
+        replaceString = '%sMesh {\n' % name
+        self.shapeIndex += 1
+        return replaceString
+
+    def convert(self, inFile, outFile=None):
+        path = os.path.dirname(inFile)
+        self.robotName = os.path.basename(inFile)
+        self.robotName = os.path.splitext(self.robotName)[0]
```

Comment:
```suggestion
        self.robotName = os.path.splitext(os.path.basename(inFile))[0]
```

I would merge these two lines as they look confusing.